### ISSUE TITLE
[HOLD for payment 2024-10-25] [$250] QAB - Destination opens in the background when starting QAB flow

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#46352]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**  9.0.13-3
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Launch New Expensify app
2. Go to workspace chat
3. Submit an expense
4. Return to LHN
5. Tap FAB
6. Tap Submit expense under QAB section
7. Tap app back button

## Expected Result:
App will return to LHN after closing QAB flow

## Actual Result:
App will return to workspace chat (destination) after closing QAB flow
The destination (in this case, workspace chat) opens in the background when starting QAB flow

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/28466cc9-9eee-4bb2-9513-8e869723b687

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01234964a7fed98878</li>
        <li>Upwork Job ID: 1819145506544874650</li>
        <li>Last Price Increase: 2024-09-26</li>
<li>Automatic offers: </li>
<ul>
<li>tsa321 | Contributor | 104236266</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @aimane-chnaif</details>

**Comment by lanitochka17** [comment]
@kevinksullivan  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by kevinksullivan** [comment]
Reproduced. I think this is a polish item for collect.

**Comment by tJonPope727** [comment]
I have seen about this issue, what do you need fix exactly?

**Comment by BrtqKr** [comment]
@aimane-chnaif I think this part was to avoid overlaying two not found views on each other, because in standard circumstances we've been getting two not found views next to each other, but when narrowing it down the screens were ending up with an overlay

**Comment by kevinksullivan** [comment]
Still figuring out the ideal solution.

**Comment by kevinksullivan** [comment]
@aimane-chnaif anything to add on what we need to do here?

**Comment by kevinksullivan** [comment]
Looping in another BZ as I'm going OOO. @aimane-chnaif , we may want to open this up if we don't hear back today/tomorrow.

**Comment by lschurr** [comment]
What's next on this one @aimane-chnaif - Are we waiting on proposals?

**Comment by mallenexpensify** [comment]
@lschurr can you share in [#expensify-swm](https://expensify.enterprise.slack.com/archives/C04878MDF34) once they post their weekly update on their projects and bandwidth? Should be tomorrow/Tues.

**Comment by lschurr** [comment]
I'm OOO until 10/7 but looking at the cal, I think @kevinksullivan is back today? Are you able to take this one back Kev?

**Comment by kevinksullivan** [comment]
posted in expensify-swm

https://expensify.slack.com/archives/C04878MDF34/p1727358544571459

**Comment by Guccio163** [comment]
Hi, I'm Wiktor Gut from SWM agency and I'll look into this 🔎

**Comment by mallenexpensify** [comment]
Thanks @Guccio163 , assigned.

**Comment by Guccio163** [comment]
Hi @aimane-chnaif, I'm Wiktor Gut from SWM agency, please assign me this issue.

**Comment by Guccio163** [comment]
I dug deep into this and I think it would be cleanest if we just added these couple of problematic screens to the `RHP_SCREENS_OPENED_FROM_LHN`, because this is the least intrusive solution; as mentioned in the comment [here](https://github.com/Expensify/App/issues/46352#issuecomment-2303854807) this issue is a result of deleting 
```
if (isNarrowLayout) {
        metainfo.isFullScreenNavigatorMandatory = false;
        metainfo.isCentralPaneAndBottomTabMandatory = false;
    }
```
in [this](https://github.com/Expensify/App/pull/41665/files) @kosmydel 's PR; he prioritised `RHP_SCREENS_OPENED_FROM_LHN` to cover for that change mentioned [here](https://github.com/Expensify/App/pull/41665/files#r1623740489).

Moreover we would have to edit and/or flag a lot of functions to detect the `Quick action` choice in FAB Pop-up, because it isn't an obvious task how to get that information from url after we navigate to RHP. It's worth mentioning that changes like that could result in further regressions and we have a simple solution to prevent them.

To sum up in my opinion it would be not only the simplest but the best to add these couple of screens to `RHP_SCREENS_OPENED_FROM_LHN`; after all they are exactly RHP screens, that we open from LHN in this situation ;)

**Comment by mallenexpensify** [comment]
Since I happen to be in this issue for some reason... 
At a high level, if a contributor provides a solution on an issue that has the `Help Wanted` label, they might be eligible for compensation.  I think my preference here would be to have @tsa321 raise the PR, since that keeps things _easy_, where they'd get paid the full amount. @aimane-chnaif is there a reason you wouldn't recommend this? (I realize you found the RCA, thanks for helping there)

**Comment by dangrous** [comment]
Jumping in here too, I think if the solution is correct let's move forward with @tsa321; it's definitely not the ideal that the root cause was incorrect originally, but now that we've determined it and the proposal still solves it in the best way possible, I think that's okay.

I'll wait a bit for any further discussion and then plan to assign @tsa321 - let's just make sure in the future we make sure to find that correct root cause since that will give us the best solution!

**Comment by Guccio163** [comment]
I see that @tsa321 will handle this one, feel free to call me for a review etc. when the PR is done though ;)

**Comment by sonialiap** [comment]
Payment summary:
- @aimane-chnaif $250 please complete the [checklist](https://github.com/Expensify/App/issues/46352#issuecomment-2423056985) and then request payment in ND
- @tsa321 $250 paid in upwork ✔️

**Comment by sonialiap** [comment]
@aimane-chnaif oh sorry! I saw that you're whitelisted for ND payments. [Offer](https://www.upwork.com/ab/applicants/1850938967757956992/hired) sent in Upwork
